### ISSUE TITLE
Incompatible with smooth scroll-behavior

### ISSUE THREAD

**Issue by simonberube** [dequelabs/axe-core#1920]
Axe doesn't identify all issues on a page if it has CSS smooth-scroll:
```css
html { scroll-behavior: smooth; }
```

I'm not sure which issues require Axe to scroll, but color contrast is one of them.

Expectation: Axe should identify issues regardless of scroll-behavior. Scroll-behavior could temporarily set to auto and then returned to its previous value. It might be more appropriate to have Axe delay analysis until each scroll position has been reached.

```css
html { scroll-behavior: auto; }
```
```javascript
document.body.parentElement.style.scrollBehavior = 'auto';
```

Actual: Axe appears to go through window scroll positions quickly, and it never reaches the destination while CSS smooth scrolling is enabled.

Motivation: (self-evident - to have all results)

<pre><code>
axe-core version: 3.3.2
axe-webdriver, extension or other integration version: v4.1.0

Browser and Assistive Technology versions

For Tooling issues:
- Platform:  Windows 10
</code></pre>

**Comment by simonberube** [comment]
How's this for a solution?

```javascript
// get initial scroll behavior
var initial_scroll_behavior = document.body.parentElement.style.scrollBehavior;

// set normal scroll behavior
document.body.parentElement.style.scrollBehavior = 'auto';

/*
  run Axe here
*/

// restore initial scroll behavior
document.body.parentElement.style.scrollBehavior = initial_scroll_behavior;
```

Depending on how Axe works, this concept might have to be repeated for all elements it scrolls through.

**Comment by WilcoFiers** [comment]
Thanks for reporting! Yup that is definitely something to consider. Axe-core only scrolls in the color-contrast rule at the moment. We have a rewrite of the color contrast check in the works that wouldn't involve scrolling, so that should hopefully fix the issue.

**Comment by padmavemulapati** [comment]
As we no longer scroll the page it doesn't affect smooth-scroll behavior, tested it with larger test file which is `https://web.archive.org/web/20180625151005/https://users.math.msu.edu/users/wongwwy/expository/torsion.html` , on running the performance of this larger fine found that its not taking more than 11s for total rendering and analyzing , confirming as everything looks good.
![image](https://user-images.githubusercontent.com/47409579/74041933-9fc7c380-49ec-11ea-92e8-4bd3434d2031.png)
